src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -75,7 +77,7 @@ public function __construct(ContainerInterface $container, MessageFormatterInter
 
         $this->options = array_merge($this->options, $options);
         $this->resourceLocales = array_keys($this->options['resource_files']);
-        $this->addResourceFiles($this->options['resource_files']);
+        $this->resourceFiles = $this->options['resource_files'];
```

Comment:
This will create a copy of $this->options['resource_files'] array, right? Wouldn't it be better for memory to have a $resourceFilesLoaded boolean attribute instead, and use $this->options['resource_files'] later to iterate at addResourceFiles?

Reply:
This will only copy the array of there is a modification to the array, if we don't touch to the array the copy-on-write mechanism will just keep a second reference to the same array.

Reply:
actually, thanks to [copy-on-write](https://en.wikipedia.org/wiki/Copy-on-write), no duplication will ever happen

Reply:
Then, is ok for me also 😀